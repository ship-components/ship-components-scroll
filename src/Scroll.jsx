/** ****************************************************************************
 * Scroll
 *
 * @author       Ryan Jung <ryan.jung@playstation.sony.com>
 * @file         Material Design Scroll
 ******************************************************************************/

// External Modules
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

// Local
import css from './scroll.css';

export default class Scroll extends React.Component {
  /**
   * Base Scroll
   * @param  {Object} props
   */
  constructor(props) {
    super(props);
  }

  /**
   * Render
   * @return {React}
   */
  render() {
    let scrollClasses = classNames(
      'scroll--shipComponentsScroll',
      css[this.props.type],
      css[this.props.orientation]
    );

    // Construct props
    let props = {
      className: scrollClasses,
      type: this.props.type,
      orientation: this.props.orientation
    }

    this.tagName = this.props.tag;

    // console.log(css);

    return (
      <div {...props}>
        {this.props.children}
      </div>
    );
  }
}

/**
 * Type Checks
 * @type {Object}
 */
Scroll.propTypes = {
  tag: PropTypes.string,
  type: PropTypes.oneOf(['shadowScroll', 'moonlightScroll']),
  orientation: PropTypes.oneOf(['horizontal', 'vertical', 'both'])

};

/**
 * Defaults
 * @type {Object}
 */
Scroll.defaultProps = {
  tag: 'div',
  type: 'shadowScroll',
  orientation: 'both'
};
