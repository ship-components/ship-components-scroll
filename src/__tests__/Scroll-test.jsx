/**
 * Test Scroll component
 */

import React from 'react';
import TestUtils from 'react-dom/test-utils';
import { shallow } from 'enzyme';
import Scroll from '../Scroll';

describe('Scroll', () => {

  it('renders without error', () => {
    let element = React.createElement(
       Scroll,
       {},
       (<div>Some child content</div>)
    );

    expect(() => TestUtils.renderIntoDocument(element))
       .not.toThrow();
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(
        <Scroll>
          <div>Some child content</div>
        </Scroll>
    );
    expect(wrapper).toMatchSnapshot();
});

});
