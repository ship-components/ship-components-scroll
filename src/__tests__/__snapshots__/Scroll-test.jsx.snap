// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scroll
    orientation="both"
    tag="div"
    type="shadowScroll"
  >
    <div>
      Some child content
    </div>
  </Scroll>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Some child content
      </div>,
      "className": "scroll--shipComponentsScroll",
      "orientation": "both",
      "type": "shadowScroll",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Some child content",
      },
      "ref": null,
      "rendered": "Some child content",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Some child content
        </div>,
        "className": "scroll--shipComponentsScroll",
        "orientation": "both",
        "type": "shadowScroll",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Some child content",
        },
        "ref": null,
        "rendered": "Some child content",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
